### PR TITLE
[X86] Put R20/R21/R28/R29 later in GR64 list

### PR DESCRIPTION
**Pull request by fzou1**
Because these registers require an extra byte to encode in certain memory form. Putting them later in the list will reduce code size when EGPR is enabled. And align the same order in GR8, GR16 and GR32 lists. Example:

    movq (%r20), %r11  # encoding: [0xd5,0x1c,0x8b,0x1c,0x24]
    movq (%r22), %r11  # encoding: [0xd5,0x1c,0x8b,0x1e]

### CODE REVIEW

**Review by phoebewang — APPROVED** [review]
LGTM.

File: llvm/lib/Target/X86/X86RegisterInfo.td [hunk 1]
```diff
@@ -550,9 +550,9 @@ def SSP : X86Reg<"ssp", 0>;
 // cannot be encoded.
 def GR8 : RegisterClass<"X86", [i8], 8,
                         (add AL, CL, DL, AH, CH, DH, BL, BH, SIL, DIL, BPL, SPL,
-                             R8B, R9B, R10B, R11B, R16B, R17B, R18B, R19B, R20B,
-                             R21B, R22B, R23B, R24B, R25B, R26B, R27B, R28B, R29B,
-                             R30B, R31B, R14B, R15B, R12B, R13B)> {
+                             R8B, R9B, R10B, R11B, R16B, R17B, R18B, R19B, R22B,
```
Comment (RKSimon) [inline llvm/lib/Target/X86/X86RegisterInfo.td:553]: Update the comment above which refers to R12/R13 being allocated last

Comment (fzou1) [reply llvm/lib/Target/X86/X86RegisterInfo.td:553]: Updated. Thank you.

**Review by KanRobert — APPROVED** [review]
LGTM

**Review by RKSimon — APPROVED** [review]
LGTM - cheers

**Pull request merged**